text/0054-kerberos-buildpack.md

```diff
@@ -0,0 +1,70 @@
+# Introduce Kerberos Buildpack
+
+## Summary
+
+Introduce a buildpack that installs the [MIT
+Kerberos](https://kerberos.org/dist/) libraries and executables such that
+applications authenticating with Kerberos can do so on any of the supported
+stacks.
+
+## Motivation
+
+Kerberos is currently supported for applications built on top of the Full
+stack. This is enabled through the addition of the `krb5-user` package in our
+stack definition. Unfortunately, the Full stack also includes a number of other
+packages that many application developers would like to avoid installing.
+
+Creating a new Kerberos buildpack will allow these developers to make a more
+focused addition to their application image be ensuring that only those
+packages relating to Kerberos are installed. It will also allow the project to
+support Kerberos on top of stacks that are not the Full stack without needing
+to also include that package in these other stack definitions.
+
+## Detailed Explanation
+
+A new repository will be created at `paketo-buildpacks/kerberos`. This repo
+will house a buildpack that has the singular purpose of installing the Kerberos
+libraries and executables into a layer made available to other buildpacks or at
+runtime. The buildpack will include Kerberos as a precompiled package that can
+be dropped into a layer with support for Jammy only. The buildpack will follow
+the existing provide/requires conventions, exposing a dependency called
+`kerberos`.
```

Comment:
I guess I wasn't just thinking a `dpkg --install` to some layer. I was expecting that we compile this from source, hopefully in a way that enables it to be run on more than just Ubuntu stacks. Additionally, I'm eventually gonna want to write an RFC that suggests we include this in some language family buildpacks. Are you suggesting that we pull buildpacks we don't maintain into those metabuildpacks?

Reply:
I think my thought is that compiling and distributing software is a lot of work, and there are entire teams of people already doing that. If we can skip that part and use what's already being offered, it gives us more time to focus on buildpacks.

My thought would be to perhaps make an official apt-buildpack under Paketo. The OSS one is getting old, and I'm not sure it's being updated (I don't think it works with Jammy). Possibly we fork from that? At any rate, we can jump to the point of being able to install the Kerberos software in the Paketo stacks much faster, IMO. Not on point for this RFC, but it also has a broader use beyond just this particular package, as you can install many other packages this way too.

As for other stacks, if we have to add an rpm-buildpack (or another package format) to support other stacks, then we could explore that. 

My only hesitation would be if there are known limitations with the existing packages, such that they won't work when installed in non-standard locations. Has anyone tried using the existing apt-buildpack and the required packages? I'm wondering if it works? or if there are known problems with that approach?

Reply:
I guess I worry that this expands the scope of what I really view as a stop-gap to address a need in the short-term. If we're talking about being able to install all sort of OS-level packages, I'd really like to see that in a stack extension, and I'd be happy to wait until we have more fleshed out support in CNB to make that happen. But, for now, I'd either like to see us implement something more limited or we can choose to do nothing and I can explore enabling these types of applications outside of Paketo.

Reply:
That's what worries me about this RFC, the stop-gap nature of this. Building, maintaining, and distributing the software isn't trivial, it also includes a buildpack to build, maintain and release. After upstream releases stack extensions we're not going to want to keep using it, but you can't release something and then deprecate it in 6 months or really even a year. Users will expect a longer-term availability, which puts the project in a bad position. Either we spend resources to keep maintaining it, or we deprecate it in short order.

If there is intended to be just a stop-gap thing, I'd rather see it go into paketo-community. That will temper user any expectations.

My thought with the apt-buildpack (and similar) is that a.) we don't have to build/maintain the software and b.) it might have some usable life even after stack extensions are available. The stack extensions RFC is not a perfect solution, it'll have advantages and disadvantages. Perhaps an apt-buildpack can fill in some of those disadvantages, but even if it doesn't the ongoing maintenance of such a buildpack is much less because we don't have to maintain the software, just the buildpack itself.

Reply:
Ok. I think this is leading towards me finding another route to get this outcome. I'm not gonna go the `apt` buildpack route. I just don't have any interest in opening up and maintaining that rabbit hole of complexity.